functions/build.sh

```
@@ -0,0 +1,54 @@
+
+# Check if running in GitHub Actions
+if [ -z "$GITHUB_ACTIONS" ]; then
+    echo "Running locally - attempting to authenticate with AWS CLI..."
+    # Get AWS account ID from the ECR URL
+    ACCOUNT_ID=$(echo "$ECR_REPOSITORY_URLS" | grep -o '[0-9]\{12\}' | head -1)
+    REGION="us-east-1"  # Hardcoded since we know we're using us-east-1
+    
+    # Login to ECR using AWS CLI
+    aws ecr get-login-password --region $REGION | docker login --username AWS --password-stdin "$ACCOUNT_ID.dkr.ecr.$REGION.amazonaws.com"
+    
+    if [ $? -ne 0 ]; then
```

Shellcheck (0.10.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)